extendeddatasquare.go

```diff
@@ -15,29 +15,34 @@ import (
 type ExtendedDataSquare struct {
 	*dataSquare
 	codec             Codec
+	treeName          string
 	originalDataWidth uint
 }
 
 func (eds *ExtendedDataSquare) MarshalJSON() ([]byte, error) {
 	return json.Marshal(&struct {
 		DataSquare [][]byte `json:"data_square"`
 		Codec      string   `json:"codec"`
+		Tree       string   `json:"tree"`
 	}{
 		DataSquare: eds.dataSquare.Flattened(),
 		Codec:      eds.codec.Name(),
+		Tree:       eds.treeName,
 	})
 }
```

Comment:
[blocking] since we're modifying how EDS's get marshalled and umarshalled from JSON, I assume this is a breaking change. If that's the case, can you please update the PR title to include an `!` to conform to conventional commits?

Proposal: "fix!: UnmarshalJSON is..."

Ref: https://www.conventionalcommits.org/en/v1.0.0/

Reply:
gotcha